server/src/main/java/org/opensearch/index/SegmentReplicationStatsTracker.java

```diff
@@ -33,6 +33,20 @@ public SegmentReplicationStatsTracker(IndicesService indicesService) {
         rejectionCount = ConcurrentCollections.newConcurrentMap();
     }
 
+    public SegmentReplicationRejectionStats getRejectionStats() {
+        long rejectionCount = 0;
+        for (IndexService indexService : indicesService) {
+            if (indexService.getIndexSettings().isSegRepEnabled()) {
+                for (IndexShard indexShard : indexService) {
+                    if (indexShard.indexSettings().isSegRepEnabled() && indexShard.routingEntry().primary()) {
+                        rejectionCount += getStatsForShard(indexShard).getRejectedRequestCount();
```

Comment:
`getStatsForShard` still fetches all per group replication stats that we don't need here.  We can get the rejection counts directly from this class's `rejectionCount` map.

we also don't need a per shard isSegRepEnabled check, the values in the rejectionCount map will only get populated for SR indices.  You can directly sum the values of the map.

```java
    public SegmentReplicationRejectionStats getTotalRejectionCount() {
        return new SegmentReplicationRejectionStats (this.rejectionCount.values().stream().mapToInt(AtomicInteger::get).sum());
    }
```